src/MoBi.Engine/Sbml/ParameterImporter.cs

```diff
@@ -7,53 +7,62 @@
 
 namespace MoBi.Engine.Sbml
 {
-    public class ParameterImporter : SBMLImporter
-    {
-        private readonly List<IEntity> _paramList;
+   public class ParameterImporter : SBMLImporter
+   {
+      private readonly List<IEntity> _paramList;
+      private UnitDefinitionImporter _unitDefinitionImporter;
 
-        public ParameterImporter(IObjectPathFactory objectPathFactory, IObjectBaseFactory objectBaseFactory, ASTHandler astHandler, IMoBiContext context)
-            : base(objectPathFactory, objectBaseFactory, astHandler, context)
-        {
-            _paramList = new List<IEntity>();
-        }
+      public ParameterImporter(IObjectPathFactory objectPathFactory, IObjectBaseFactory objectBaseFactory, ASTHandler astHandler, IMoBiContext context)
+          : base(objectPathFactory, objectBaseFactory, astHandler, context)
+      {
+         _paramList = new List<IEntity>();
+      }
 
-        protected override void Import(SBMLModel model)
-        {
-            for (long i = 0; i < model.getNumParameters(); i++)
-            {
-                _paramList.Add(CreateParameter(model.getParameter(i)));
-            }
-            AddToProject();
-        }
+      protected override void Import(SBMLModel model)
+      {
+         for (long i = 0; i < model.getNumParameters(); i++)
+         {
+            _paramList.Add(CreateParameter(model.getParameter(i)));
+         }
+         AddToProject();
+      }
 
-        /// <summary>
-        ///     Creates a MoBi Parameter from a given SBML Parameter.
-        /// </summary>
-        public IFormulaUsable CreateParameter(Parameter sbmlParameter)
-        {
-            var value = 0.0;
-            if (sbmlParameter.isSetValue()) value = sbmlParameter.getValue();
-            IFormula formula = ObjectBaseFactory.Create<ConstantFormula>().WithValue(value);
+      public void SetUnitDefinitionImporter(UnitDefinitionImporter unitDefinitionImporter)
+      {
+         _unitDefinitionImporter = unitDefinitionImporter;
```

Comment:
Why is the service not injected in constructor?

Reply:
There were no clear interfaces for that. I refactor some parts of the code and we can inject the importers but in general the SBML Importer old code is not following some principles of inversion of control. I think we should not address this now but maybe later we should revisit the topic.